universe/ValueRef.h

```diff
@@ -58,6 +62,12 @@ struct FO_COMMON_API ValueRef : public ValueRefBase
       * to objects such as the source, effect target, or condition candidates
       * that exist in the tree. */
     virtual T Eval(const ScriptingContext& context) const = 0;
+
+    /** Evaluates the expression tree with an empty context and retuns the
+      * a string representation of the result value iff the result type is
+      * supported (currently std::string, int, float, double, enum).
+      * See ValueRefs.cpp for specialisation implementations. */
+    std::string StringResult() const;
```

Comment:
mark as `override` ?

Reply:
possibly also `final` if it makes sense

Reply:
Maybe also clarify how the string is expected to be formatted and used for non-trivial cases. Eg. for enumerations and names of content, is it going to return something like `PT_BARREN` or pre-localized "Barren", and `SH_NANOROBOTIC` or "Nano-Robotic Hull" for whatever stringtable is set where it's being evaluated? (I assume it will `PT_BARREN` type non-localized string representations.)

Reply:
Enums are currently returned using the underlying value e.g. for Uninhabitable: `0` as string

Reply:
Whatever is returned for enums should be usable by AI in the end. I wait until a use case comes up.

Made it `final`